lib/config.js

```diff
@@ -738,19 +738,18 @@ util.loadFileConfigs = function(configDir, options) {
  */
 util.locateMatchingFiles = function(configDirs, allowedFiles) {
   return configDirs.split(Path.delimiter)
+    .filter(dir => dir)
     .reduce(function(files, configDir) {
-      if (configDir) {
-        configDir = _toAbsolutePath(configDir);
-        try {
-          FileSystem.readdirSync(configDir).forEach(function(file) {
-            if (allowedFiles[file]) {
-              files.push([allowedFiles[file], Path.join(configDir, file)]);
-            }
-          });
-        }
-        catch(e) {}
-        return files;
```

Comment:
This returns null if any of the configDir return values are falsey.  This is what the stalled PR was trying to fix.

Reply:
`.filter(dir => dir)` is tricky to read. I recommend clearer syntax for posterity:

```
.filter(dir => Boolean(dir))
```

This makes the implicit type-casting explicit. Yes, it's unnecessary for the JavaScript interpreter, but it's clearer for humans to read.